File: groupAnagrams.py
class Solution:
    def groupAnagrams(self, strs: list[str]):

        sorted_list = []
        size_set = []
        for i in range(len(strs)):
            size_set.append("".join(sorted(strs[i])))
        new_set = set(size_set)
        print(new_set)
        for i in range(len(new_set)):
            new_list = []
            for j in range(0,len(strs)):
                word_j = strs[j]
                sorted_i = list(new_set)[i]
                sorted_j = "".join(sorted(word_j))
                if sorted_i == sorted_j and word_j not in sorted_list:
                    print("Word = ", word_j, " <- index no = ", j, end=" ")
                    new_list.append(word_j)
            sorted_list.append(new_list)
            print()
        return sorted_list

def main():
    print(Solution().groupAnagrams(["eat","tea","tan","ate","nat","bat"]))

if __name__ == "__main__":
    main()

File: java/Pyhton_Conversion/groupAnagrams.java
package Pyhton_Conversion;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class groupAnagrams {
    public List<List<String>> groupAnagramsFunciton(String[] strs) {

        Set<String> inputSet = new HashSet<>();
        for (int i = 0; i < strs.length; i++) {
            inputSet.add(strs[i]);
        }
        System.out.println("Inputt set " + inputSet);
        List<List<String>> li = new ArrayList<>();
        return li;
    }

    public static void main(String[] args) {
        groupAnagrams obj = new groupAnagrams();
        String[] strs = { "eat", "tea", "tan", "ate", "nat", "bat", "eat" };
        System.out.println(obj.groupAnagramsFunciton(strs));
    }
}
